src/main/java/com/criticalsoftware/AnnouncementResource.java

```diff
@@ -0,0 +1,148 @@
+import jakarta.validation.ConstraintViolationException;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.bson.types.ObjectId;
+
+@Path("/announcements")
+public class AnnouncementResource {
+
+    @Inject
+    private AnnouncementRepository announcementRepository;
+
+    @Inject
+    private AnnouncementService announcementService;
+
+    @Inject
+    private UserRepository userRepository;
+
+    // Create an announcement
+    @POST
+    public Response create(@Valid AnnouncementRequest request) {
+        try {
+            User userDonor = userRepository.findById(new ObjectId(request.getUserDonorId()));
+            if (userDonor == null) {
+                return Response.status(Status.BAD_REQUEST)
+                        .entity("User donor ID does not exist.")
+                        .build();
+            }
+
+            if (isProductFieldsEmpty(request)) {
+                return Response.status(Status.BAD_REQUEST)
+                        .entity("All product fields must be filled.")
+                        .build();
+            }
+
+            Product product = new Product(
+                    request.getProductName(),
+                    request.getProductDescription(),
+                    request.getProductPhotoUrl(),
+                    request.getProductCategory()
+            );
+
+            Announcement announcement = new Announcement(
+                    product,
+                    userDonor.getId()
+
+            );
+
+            announcement.setId(new ObjectId().toString());
+
+            announcementRepository.persist(announcement);
+
+            Map<String, Object> responseMap = new HashMap<>();
+            responseMap.put("message", "Announcement created successfully with ID: " + announcement.getId());
+            responseMap.put("announcement", new AnnouncementResponse(
+                    announcement.getId(),
+                    announcement.getProduct(),
+                    announcement.getUserDonorId(),
+                    announcement.getUserDoneeId(),
+                    announcement.getDate()
+            ));
+
+            return Response.created(new URI("/announcements/" + announcement.getId()))
+                    .entity(responseMap)
+                    .build();
+        } catch (ConstraintViolationException e) {
+            String errorMessages = e.getConstraintViolations().stream()
+                    .map(violation -> violation.getPropertyPath() + ": " + violation.getMessage())
+                    .collect(Collectors.joining(", "));
+            return Response.status(Status.BAD_REQUEST)
+                    .entity("Validation error: " + errorMessages)
+                    .build();
+        } catch (Exception e) {
+            return Response.status(Status.INTERNAL_SERVER_ERROR)
+                    .entity("Error in processing the request: " + e.getMessage())
```

Comment:
As you use this string several times in this class, you can replace it with a constant.